src/services/index.ts

```diff
@@ -44,8 +44,20 @@ export function createLookupList<K, V>(entries: [K, V][]): LookupList<K, V> {
   };
 }
 
-export function throwIfNotOk(response: Response) {
-  if (!response.ok) throw new Error(response.statusText);
+// Define a custom error class to include the response and status code
+export class ResponseError extends Error {
+  constructor(
+    message: string,
+    public response: Response,
+  ) {
+    super(message);
+  }
+}
+
+export function throwIfNotOk(response: Response): Response & { ok: true } {
```

Comment:
Nice solution, better then having just the statusText. This does mean we have to check if there is any code depending on the error with only the statusText